torch/csrc/jit/passes/hoist_conv_packed_params.cpp

```diff
@@ -0,0 +1,128 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/hoist_conv_packed_params.h>
+#include <torch/csrc/jit/passes/quantization/helper.h>
+
+namespace torch {
+namespace jit {
+
+// Hoists packed params from a conv module to the parent module.
+// The benefit is that after this hoisting, the conv module
+// no longer holds anything and can be deleted, reducing model
+// size.
+//
+// Before (easy case):
+//
+// %1 = prim::GetAttr[name="conv1"][%self]
+// %2 = prim::GetAttr[name="_packed_params][%1]
+//
+// After (easy case):
+//
+// %2 = prim::GetAttr[name="{prefix}.conv1._packed_params"][%self]
+//
+// Before (generic case):
+//
+// %1 = prim::GetAttr[name="name1"][%self]
+// %2 = prim::GetAttr[name="name2"][%1]
+// ...
+// %n = prim::GetAttr[name="_packed_params][%n-1]
+//
+// After (generic case):
+//
+// %n =
+// prim::GetAttr[name="{prefix}.name1{...}.name(n-1)._packed_params"][%self]
+//
+void hoistConvPackedParams(
```

Comment:
Can we also add a pass for linear? Is there value in doing this for linear modules too?

Reply:
I don't see why not, although I'd defer to the mobile team on the priority.